### PR TITLE
#515 [HxChipList] Support for Reset button (Remove all)

### PR DESCRIPTION
**Pull request by crdo**
#515

### CODE REVIEW

File: Havit.Blazor.Components.Web.Bootstrap/Chips/HxChipList.razor [hunk 1]
```diff
@@ -9,11 +9,20 @@
...
         }
+        @if (Reset == true)
+        {
+            @ResetButtonTemplate
+            <span role="button" class="hx-chip-reset-btn"
+                @onclick="() => HandleResetClick()" @onclick:stopPropagation>
+                @ResetButtonText
+                <HxIcon Icon="BootstrapIcon.X" />
+            </span>
+        }
```
Comment (hakenr) [inline Havit.Blazor.Components.Web.Bootstrap/Chips/HxChipList.razor:26]: How do I use `ResetButtonTemplate`?
I probably do not need to have ability to trigger the `OnResetClick` from the template (the handler can be used directly), but I will probably want to get rid of the original `span.hx-chip-reset-btn`.

+ the class should be `.hx-chip-list-reset-btn` (missing the `-list-` part)

File: Havit.Blazor.Components.Web.Bootstrap/Chips/HxChipList.razor.cs [hunk 2]
```diff
@@ -64,6 +64,15 @@ static HxChipList()
 	/// Called when chip remove button is clicked.
 	/// </summary>
 	[Parameter] public EventCallback<ChipItem> OnChipRemoveClick { get; set; }
+
+	[Parameter] public EventCallback<ChipItem> OnResetClick { get; set; }
+
+	[Parameter] public bool Reset { get; set; }
```
Comment (hakenr) [inline Havit.Blazor.Components.Web.Bootstrap/Chips/HxChipList.razor.cs:70]: Should I add the `Settings` + `Defaults` to be able to pre-set these?

File: Havit.Blazor.Components.Web.Bootstrap/Chips/HxChipList.razor.cs [hunk 3]
```diff
@@ -73,4 +82,14 @@ private async Task HandleRemoveClick(ChipItem chipItemToRemove)
...
+
+	/// <summary>
+	/// Triggers the <see cref="OnResetClick"/> event. Allows interception of the event in derived components.
+	/// </summary>
+	protected virtual Task InvokeOnResetClickAsync() => OnResetClick.InvokeAsync();
+
+	private async Task HandleResetClick()
+	{
+		await InvokeOnResetClickAsync();
+	}
```
Comment (hakenr) [inline Havit.Blazor.Components.Web.Bootstrap/Chips/HxChipList.razor.cs:94]: You can use `InvokeOnResetClickAsync` from the markup directly, probably no need to have this pass-through method.

File: Havit.Blazor.Components.Web.Bootstrap/Chips/HxChipList.razor.css [hunk 4]
```diff
@@ -8,7 +8,7 @@
     align-items: center;
 }
 
-::deep .hx-chip-remove-btn {
+ .hx-chip-remove-btn {
```
Comment (hakenr) [inline Havit.Blazor.Components.Web.Bootstrap/Chips/HxChipList.razor.css:11]: `.hx-chip-list-remove-btn`?

**Review by hakenr — CHANGES_REQUESTED** [review]

File: Havit.Blazor.Components.Web.Bootstrap/Chips/HxChipList.razor.cs (hunk 2 above)
Comment (crdo) [reply Havit.Blazor.Components.Web.Bootstrap/Chips/HxChipList.razor.cs:70]: @hakenr yes please.

**Comment by hakenr** [comment]
@crdo I think the _Reset_ button should not be displayed when there are no chips.
I will take care of that (if it is already not there "by design").

**Pull request merged**